### PR TITLE
[SYCL][HIP][L0][NATIVECPU] `urAdapterGetLastError` fix so messages aren't lost.

### PR DESCRIPTION
**Pull request by JackAKirk**
`urAdapterGetLastError` is being called in l0 and hip plugins but just returning success, which means the messages were lost. I've implemented `urAdapterGetLastError` in these backends in the same way as in the CUDA backend (https://github.com/intel/llvm/pull/10626) to fix this. I also made the same change for nativeCPU: this backend does not yet call `urAdapterGetLastError` but it could in the future.

Fixed hip message:

adapters/hip/enqueue.cpp:        setErrorMessage(LocalMemSzPtrUR ? "Invalid value specified for "

Fixed l0 messages:

adapters/level_zero/kernel.cpp:    setErrorMessage("Write device global variable is out of range.",
adapters/level_zero/kernel.cpp:    setErrorMessage("Read from device global variable is out of range.",
adapters/level_zero/device.cpp:      setErrorMessage("Set ZES_ENABLE_SYSMAN=1 to obtain free memory",

The associated errors returned by UR for the adapters/level_zero/kernel.cpp have also been fixed: SYCL runtime will only retrieve the errors if `UR_RESULT_ERROR_ADAPTER_SPECIFIC` is returned.

### CODE REVIEW

File: sycl/plugins/unified_runtime/ur/adapters/level_zero/adapter.cpp [hunk 1]
```diff
@@ -180,10 +180,9 @@ UR_APIEXPORT ur_result_t UR_APICALL urAdapterGetLastError(
                    ///< error code will be stored.
 ) {
   std::ignore = Adapter;
-  std::ignore = Message;
   std::ignore = Error;
-  urPrint("[UR][L0] %s function not implemented!\n", __FUNCTION__);
-  return UR_RESULT_ERROR_UNSUPPORTED_FEATURE;
+  *Message = ErrorMessage;
```
Comment (jandres742) [inline sycl/plugins/unified_runtime/ur/adapters/level_zero/adapter.cpp:184]: thanks @JackAKirk . L0 implementation for getLastError is being done here https://github.com/intel/llvm/pull/10784. There's a discussion here on how it should behave. i'm planning on following on that, so feel free to address the hip and cpu adapters only on this patch, or to address also L0 based on those comments. Either way works for me.

**Review by PietroGhg — APPROVED** [review]
Change looks good for Native CPU since it aligns our implementation with the other adaptors, thank you :+1:

File: sycl/plugins/unified_runtime/ur/adapters/level_zero/adapter.cpp (hunk 1 above)
Comment (JackAKirk) [reply sycl/plugins/unified_runtime/ur/adapters/level_zero/adapter.cpp:184]: Sure, L0 impl here is already consistent with those comments (all backends have the same impl).

Comment (JackAKirk) [reply sycl/plugins/unified_runtime/ur/adapters/level_zero/adapter.cpp:184]: @jandres742 do you want me to remove the l0 implementation from this PR, or are you happy with it?

Thanks

Comment (jandres742) [reply sycl/plugins/unified_runtime/ur/adapters/level_zero/adapter.cpp:184]: @JackAKirk : implementation here is good, thanks!

**Review by jandres742 — APPROVED** [review]

**Comment by JackAKirk** [comment]
@sergey-semenov Would you be able to review this please?

Thanks

File: sycl/plugins/unified_runtime/ur/adapters/level_zero/adapter.cpp (hunk 1 above)
Comment (JackAKirk) [reply sycl/plugins/unified_runtime/ur/adapters/level_zero/adapter.cpp:184]: OK thanks!

**Comment by jandres742** [comment]
Adapters have been moved to UR repo. I'm adding the L0 changes here https://github.com/oneapi-src/unified-runtime/pull/1033

**Pull request closed without merging**